querysql/querysql.go

```diff
@@ -168,67 +168,64 @@ func Next(rs *ResultSets, scanner Target) error {
 		return rs.Err
 	}
 
-	// When we enter this function one of the following will be the case:
-	// 1) We are at the beginning of using rs.Rows
-	// 2) rs.Rows.NextResultSet() was called, returned true, and there was another result set ready for us
-	// 3) rs.Rows.NextResultSet() was called, returned false, and we have done a close and rs.Rows is nil
-
-	success := false
```

Comment:
This change is not part of the bug fix.  The change also does not modify externally observable behavior.  The change is simply for readability.

Instead, this change is a simplification.  Instead of the variable and the deferred function that keeps track of this variable's value to determine whether the `close` operation should be performed, we create a deferred function with the close when it's needed.